Source/santactl/SNTCommand.h

```diff
@@ -0,0 +1,64 @@
+/// Copyright 2017 Google Inc. All rights reserved.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///    http://www.apache.org/licenses/LICENSE-2.0
+///
+///    Unless required by applicable law or agreed to in writing, software
+///    distributed under the License is distributed on an "AS IS" BASIS,
+///    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+///    See the License for the specific language governing permissions and
+///    limitations under the License.
+
+@import Foundation;
+
+@class SNTXPCConnection;
+
+@protocol SNTCommandProtocol <NSObject>
+
+///
+///  @return YES if command requires root.
+///
++ (BOOL)requiresRoot;
+
+///
+///  @return YES if command requires connection to santad.
+///
++ (BOOL)requiresDaemonConn;
+
+///
+///  A small summary of the command, to be printed with the list of available commands
+///
++ (NSString *)shortHelpText;
+
+///
+///  A longer description of the command when the user runs <tt>santactl help x</tt>
+///
++ (NSString *)longHelpText;
+
+///
+///  Called when the user is running the command
+///  @param arguments an array of arguments passed in
+///  @param daemonConn connection to santad. Will be nil if connection failed or
+///      if @c requiresDaemonConn is @c NO
+///
+///  @note This method (or one of the methods it calls) is responsible for calling exit().
+///
++ (void)runWithArguments:(NSArray *)arguments daemonConnection:(SNTXPCConnection *)daemonConn;
+
+@end
+
+@interface SNTCommand : NSObject<SNTCommandProtocol>
```

Comment:
This super class shouldn't need to subscribe to SNTCommandProtocol. Instead the individual concrete commands should subscribe.